packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1446 @@
+  Timer interactionTimer;
+
+  @override
+  void initState() {
+    super.initState();
+    overlayController = AnimationController(
+      duration: kRadialReactionDuration,
+      vsync: this,
+    );
+    valueIndicatorController = AnimationController(
+      duration: valueIndicatorAnimationDuration,
+      vsync: this,
+    );
+    enableController = AnimationController(
+      duration: enableAnimationDuration,
+      vsync: this,
+    )..value = widget.onChanged != null ? 1.0 : 0.0;
+    startPositionController = AnimationController(
+      duration: Duration.zero,
+      vsync: this,
+      value: widget.values.start
+    );
+    endPositionController = AnimationController(
+      duration: Duration.zero,
+      vsync: this,
+      value: widget.values.end
+    );
+    enableController.value = widget.onChanged != null ? 1.0 : 0.0;
+    startPositionController.value = _unlerp(widget.values.start);
+    endPositionController.value = _unlerp(widget.values.end);
+  }
+
+  @override
+  void dispose() {
+    interactionTimer?.cancel();
+    overlayController.dispose();
+    valueIndicatorController.dispose();
+    enableController.dispose();
+    startPositionController.dispose();
+    endPositionController.dispose();
+    super.dispose();
+  }
+
+  void _handleChanged(RangeValues values) {
+    assert(widget.onChanged != null);
+    final RangeValues lerpValues = _lerpRangeValues(values);
+    if (lerpValues != widget.values) {
+      widget.onChanged(lerpValues);
+    }
+  }
+
+  void _handleDragStart(RangeValues values) {
+    assert(widget.onChangeStart != null);
+    widget.onChangeStart(_lerpRangeValues(values));
+  }
+
+  void _handleDragEnd(RangeValues values) {
+    assert(widget.onChangeEnd != null);
+    widget.onChangeEnd(_lerpRangeValues(values));
+  }
+
+  // Returns a number between min and max, proportional to value, which must
+  // be between 0.0 and 1.0.
+  double _lerp(double value) {
+    assert(value >= 0.0);
+    assert(value <= 1.0);
+    return value * (widget.max - widget.min) + widget.min;
+  }
+
+  // Returns a new range value with the start and end lerped.
+  RangeValues _lerpRangeValues(RangeValues values) {
+    return RangeValues(_lerp(values.start), _lerp(values.end));
+  }
+
+  // Returns a number between 0.0 and 1.0, given a value between min and max.
+  double _unlerp(double value) {
+    assert(value <= widget.max);
+    assert(value >= widget.min);
+    return widget.max > widget.min ? (value - widget.min) / (widget.max - widget.min) : 0.0;
+  }
+
+  // Returns a new range value with the start and end unlerped.
+  RangeValues _unlerpRangeValues(RangeValues values) {
+    return RangeValues(_unlerp(values.start), _unlerp(values.end));
+  }
+
+  // Finds closest thumb. If the thumbs are close to each other, wait for movement to select.
+  static final RangeThumbSelector _defaultRangeThumbSelector = (TextDirection textDirection, RangeValues values, double tapValue, Size thumbSize, Size trackSize, double dx) {
```

Comment:
I think this would be more readable with a trailing comma and each parameter on its own line

Reply:
Done